core/templates/services/search.service.ts

```diff
@@ -13,218 +13,259 @@
 // limitations under the License.
 
 /**
- * @fileoverview search service for activityTilesInfinityGrid
+ * @fileoverview Search service for activityTilesInfinityGrid.
  */
 
-require('services/services.constants.ajs.ts');
-
-import { EventEmitter } from '@angular/core';
-
-angular.module('oppia').factory('SearchService', [
-  '$http', '$log', '$translate', 'SEARCH_DATA_URL',
-  function($http, $log, $translate, SEARCH_DATA_URL) {
-    var _lastQuery = null;
-    var _lastSelectedCategories = {};
-    var _lastSelectedLanguageCodes = {};
-    var _searchCursor = null;
-    var _isCurrentlyFetchingResults = false;
-    var numSearchesInProgress = 0;
-    var _searchBarLoadedEventEmitter = new EventEmitter();
-    var _initialSearchResultsLoadedEventEmitter = new EventEmitter();
-    // Appends a suffix to the query describing allowed category and language
-    // codes to filter on.
-    var _getSuffixForQuery =
-      function(selectedCategories, selectedLanguageCodes) {
-        var querySuffix = '';
-
-        var _categories = '';
-        for (var key in selectedCategories) {
-          if (selectedCategories[key]) {
-            if (_categories) {
-              _categories += '" OR "';
-            }
-            _categories += key;
-          }
-        }
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable, EventEmitter } from '@angular/core';
+
+import { ExplorationSummaryBackendDict } from 'domain/summary/exploration-summary-backend-api.service';
+import { SearchBackendApiService } from './search-backend-api.service';
+
+export class SelectionList {
+  [key: string]: boolean;
+}
+
+class FilterDetails {
+  description: string;
+  itemsName: string;
+  masterList: {
+    id: string;
+    text: string;
+  }[];
+  selections: SelectionList;
+  numSelections: number;
+  summary: string;
+}
+
+export class SelectionDetails {
```

Comment:
@TheoLipeles, FYI, I'll talk to angular team about dict-class and review the PR by tomorrow.